docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,200 @@
+
+Should we wish to provide a multi-tenancy model in our tenanted solution, we are free to do so in any way we choose, to meet our operational, legal, and/or regulatory obligations. We are not bound to the tenancy model of our parent.
+
+A common challenge when designing a multi-tenanted system is thinking about the right "scope". Am I thinking about the solution in my tenant (which should not expose details of tenancy) or the multi-tenancy provider itself (which should not depend on implementation details of the solutions that happen to be deployed as its tenants)? You must take care that one does not bleed into the other.
+
+## Solution plane and management plane
+You may have noted that I said "it is usually irrelevant that any given tenant has a parent". There is one respect in which the child tenant *is* aware of the existence of a parent. It needs to be able to *manage its tenancy*. To do so, it will provision, configure, and monitor services through tools provided by the parent that hosts it.
+
+Those tools may also provide means to grant temporary and/or limited access to the child tenant for identities in the parent tenant, in order to deliver support.
+
+We often refer to the divide between the services you *consume in your tenant* and the services you use to *manage your tenant* as the *solution plane* and the *management plane*.
+
+[Solution v. Management Plane](Solution v Management Plane)
+
+Careful conceptual separation of solution plane and management plane is essential.
+
+## Common requirements of a multi-tenanted solution
+Let’s think about a very generalised application. We might say that a typical application has a stimulus/response design like this:
+
+1. Receive a stimulus of some kind (e.g. HTTP request, event, message, timer)
+1. Connect to and authenticate against one or more services within or outside the solution using appropriate credentials, in order to obtain the information required to respond to the stimulus.
+1. Process that information (and optionally respond), possibly connecting to other services within or outside the solution, using appropriate credentials, to update state related to that request.
+
+There are numerous services you can use with that basic pattern – HTTP requests, queues and message busses, event stores, various flavours of databases and storage, caching. The details of these services is outside the scope of this discussion, but they are all consumed using this basic pattern.
+
+In a multi-tenanted solution, you need to layer a notion of "the intended tenant" into your design:
+
+1. Receive a stimulus of some kind, and dispatch it to the appropriate tenant
```

Comment:
"dispatch it to the appropriate tenant" makes it sound like a tenant is a distinct reachable thing, like an endpoint. That won't always be true. I the essential part here is determining which tenant(s) the stimulus is relevant to. And I tend to think of it more like "dispatch the stimulus to an appropriate handler, supplying a context identifying the appropriate tenant".

My thinking here is that in a lot of cases the target to which the stimulus gets dispatched is the same every single time. (E.g., imagine a simple multi-tenant service which offers just a single web API endpoint.) There isn't really a sense in which we dispatch to "a tenant". We dispatch to the one and only service, but we provide that service with a context enabling it to act for the right tenant.

Reply:
That's a much better expression. I have also added the "dispatch the stimulus to an appropriate handler" phrase to the non-multi-tenant example.